readme.markdown

```diff
@@ -3,44 +3,80 @@ CouchDB CI Setup
 
-Current state:
+The current (rough) plan for the build matrix is this:
 
-- [x] install bare Jenkins master with Ansible
-- [x] install and configure nginx
-- [x] create CouchDB build job in Jenkins via Ansible
-- [x] switch to master-worker Jenkins setup
-- [x] use ntp server for master and workers
-- [ ] Use SCM sync plug-in to manage job configs
-    * http://stackoverflow.com/questions/27138043/jenkins-scm-sync-configuration-plugin-in-docker-wont-talk-to-github
-    * https://cburgmer.wordpress.com/2013/01/02/tracking-configuration-changes-in-jenkins/
-- [ ] enable auth for Jenkins
-- [ ] actually fetch CouchDB from VCS
-- [ ] all apt-get commands should pin a specific version, in the base box definition as well as in Ansible. How?
-- [ ] create an additional Ubuntu worker with an older Erlang version
-- [ ] create another base box (different linux distro) for a third worker
-- [ ] talk to Infra people
+**OS/Erlang**       | **R14B04** | **R16B03-1** | **17.5** | **18.x**
+--------------------|------------|--------------|----------|---------
+**Ubuntu 14.04**    | ?          | -            | -        | WIP
+**Ubuntu latest ?** | ?          | -            | -        | -
+**Debian 7**        | ?          | -            | -        | -
+**Debian 8**        | ?          | -            | -        | -
+**OS X latest**     | ?          | -            | -        | -
+**Free BSD**        | ?          | -            | -        | -
+**Windows**         | ?          | -            | -        | -
 
-*Remark: Throughout this repository we use the terms "master"/"worker" for the Jenkins build machines, whereas the Jenkins documentation uses the terms "master"/"slave".*
+### Open questions
+
+* AFAIK Erlang 14 support will be dropped soon-ish, so I'm not sure if it is worth the effort to do anything for that.
+* Which 18.x Erlang version is to be used? I heard someone saying 18.0 once, but that was before 18.1 and 18.2 were available, so I guess it makes more sense to always use the latest 18.x to see if changes in Erlang 18 breaks CouchDB.
+* There is no CentOS/RHEL there, shouldn't it be added?
+* Do we run a CouchDB build on all combinations on each commit? This would probably be too much for the ASF Infra build systems. Do we build them once a day? We need to find a good balance between early feedback and resource consumption here.
+* Do we even want to build the master branch or some other branch/tag? I guess the master branch would be most interesting for now, but not entirely sure. Also, it might make sense to make the branch/tag parameterizable so we could also use this to create releases from a specific tag etc.
+* What exactly do we do in each Jenkins build? Just build CouchDB? Also build docs? Start CouchDB? Run some test suite?
```

Comment:
build couchdb, build docs, run all test suites.

The 'build docs' step is very onerous, taking quite a while. If possible, we'd avoid building docs again if they haven't changed since last time, though that might be difficult to achieve.